internal/pkg/http/utils.go

```diff
@@ -117,33 +125,42 @@ func ToHTTPRequest(requestString string, compression CompressionType) (Request,
 		Method:  method,
 		Headers: headers,
 		Path:    path,
-		Body:    reader,
+		Body:    &compressedBody,
 	}, nil
 }
 
-func compressGzip(data []byte) io.Reader {
-	pr, pw := io.Pipe()
```

Comment:
Optimization not required as mittens requests are re-used in the channel. So compression is only done once for each request element.